crucible-llvm/src/Lang/Crucible/LLVM/MemModel.hs

```diff
@@ -1020,16 +1020,18 @@ isAllocatedAlignedPointer sym w alignment mutability ptr size mem =
 --
 --   The pointer to read from must be concrete and nonnull.  The contents
 --   of the string may be symbolic; HOWEVER, this function will not terminate
---   until it eventually reaches a concete null-terminator or a load error.
+--   until it eventually reaches a concete null-terminator or a load error,
+--   unless a maximum bound is provided via the @gas@ parameter.
 strLen :: forall sym wptr.
   (IsSymInterface sym, HasPtrWidth wptr, Partial.HasLLVMAnn sym) =>
   sym ->
+  Maybe Word {-^ bound on loop/string length -} ->
```

Comment:
I'll be curious to hear what Rob thinks, but I like this idea in general. I'd say I'd like a newtype for `Gas` if we do this

Reply:
No strong opinion about a `Gas` newtype.